AIPscan/Data/data.py

```diff
@@ -292,3 +293,123 @@ def largest_files(storage_service_id, file_type=None, limit=20):
         report[FIELD_FILES].append(file_info)
 
     return report
+
+
+def _aips_by_format_query(storage_service_id, file_format):
+    """Fetch information on all AIPs with given format from database
+
+    This is separated into its own helper function to aid in testing.
+
+    :param storage_service_id: Storage Service ID (int)
+    :param file_format: File format (str)
+
+    :returns: SQLAlchemy query results
+    """
+    aips = (
+        db.session.query(
+            AIP.id.label("id"),
+            AIP.transfer_name.label("name"),
+            AIP.uuid.label("uuid"),
+            db.func.count(File.id).label("file_count"),
+            db.func.sum(File.size).label("total_size"),
```

Comment:
See comment in main thread about reusing the existing endpoints vs introducing new ones.

On the naming of these labels, I see your point and will do some thinking about e.g. using our existing `FIELD_X` constants rather than string literals here (at least outside of the templates, though there may be a way of addressing that).

Reply:
Hey @ross-spencer, coming back to this one a bit - I'm having a hard time replacing this string literals because of how they're only used within the scope of the Data endpoint and accessed as properties of query results.

To illustrate what I mean, we have these labels in `_query_aips_by_file_format_or_puid` (as of https://github.com/artefactual-labs/AIPscan/pull/76/commits/e12649da84d9b80ad882708810758dea5601f734):

```
aips = (
    db.session.query(
        AIP.id.label("id"),
        AIP.transfer_name.label("name"),
        AIP.uuid.label("uuid"),
        db.func.count(File.id).label("file_count"),
        db.func.sum(File.size).label("total_size"),
    )
    ...
```
These labels are only accessed in `_aips_by_file_format_or_puid` here as properties of the query results:

```
...
for result in results:
    aip_info = {}

    aip_info["id"] = result.id
    aip_info[FIELD_AIP_NAME] = result.name
    aip_info[FIELD_UUID] = result.uuid
    aip_info[FIELD_COUNT] = result.file_count
    aip_info[FIELD_SIZE] = result.total_size

    report[FIELD_AIPS].append(aip_info)

return report
```
Since the `FIELD_` constants are being used for the Data endpoint's return value, that's all the API or Reporter blueprints will see - the labels aren't referred to again after this point.

I could make constants for the SQLAlchemy labels, but AFAICT the only place they'd be used is within `_query_aips_by_file_format_or_puid`. Another option would be to remove the labels from the query, in which case we could modify the `_aips_by_file_format_or_puid` code to access data by index - something like this, which I find less readable:

```
...
for result in results:
    aip_info = {}

    aip_info["id"] = result[0]
    aip_info[FIELD_AIP_NAME] = result[1]
    aip_info[FIELD_UUID] = result[2]
    aip_info[FIELD_COUNT] = result[3]
    aip_info[FIELD_SIZE] = result[4]

    report[FIELD_AIPS].append(aip_info)

return report
```

Given all that I'm tempted to leave this as-is. Does that make sense to you?

Reply:
I see what you mean Tessa and I appreciate you following the logic through for the demonstration here. I think you are right. 

My only other suggestion is it would be nice to do the following:
```python
    AIP_ID = "id"
    TRANSFER_NAME = "name"
    AIP_UUID = "uuid"
    FILE_COUNT = "file_count"
    FILE_SIZE = "total_size"
    aips = (
        db.session.query(
            AIP.id.label(AIP_ID),
            AIP.transfer_name.label(TRANSFER_NAME),
            AIP.uuid.label(AIP_UUID),
            db.func.count(File.id).label(FILE_COUNT),
            db.func.sum(File.size).label(FILE_SIZE),
        )
        .join(File)
        .join(StorageService)
        .filter(StorageService.id == storage_service_id)
        .group_by(AIP.id)
        .order_by(db.func.count(File.id).desc(), db.func.sum(File.size).desc())
    )
```
This might be colored by my experience of some of the larger code-bases we're working with, because this quite a nice small function - but my usual rationale it's a little cleaner and lends itself better to refactor/extensibility to specify the literals as consts/variables. Any other benefits are minor, but I recognize there is some pragmatism nesting the literals in-line.